### ISSUE TITLE
I got a Error = [ GLFW Error: X11: The DISPLAY environment variable is missing ]

### ISSUE THREAD

**Issue by wooheaven** [adioshun/gitBook_PCL#1]
I'm a new comer for Open3D.

I just find out that this repository.
I decide to learn Open3D with the repository.

I create docker env for learn.
And I execute by belows but I got a error.

Do you have any support ?

```
what i execute is below
open3d.ipynb / cell[6] / draw_geometries([pcd])
```

```{bash}
ubuntu@ubuntu:~/Documents/docker/99_Open3D$ docker run -it -p 7777:8888 --name "Open3D" adioshun/ubuntu16:Open3D /bin/bash
root@79b4d73d2901:~# cp open3d.ipynb sample_table.pcd /workspace/
root@79b4d73d2901:~# jupyter notebook --allow-root
[W 08:29:11.838 NotebookApp] WARNING: The notebook server is listening on all IP addresses and not using encryption. This is not recommended.
[I 08:29:11.847 NotebookApp] Serving notebooks from local directory: /workspace
[I 08:29:11.847 NotebookApp] 0 active kernels
[I 08:29:11.847 NotebookApp] The Jupyter Notebook is running at:
[I 08:29:11.847 NotebookApp] http://[all ip addresses on your system]:8888/
[I 08:29:11.847 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[I 08:29:14.865 NotebookApp] 302 GET / (192.168.8.122) 0.96ms
[I 08:29:14.871 NotebookApp] 302 GET /tree? (192.168.8.122) 1.59ms
[I 08:29:19.743 NotebookApp] 302 POST /login?next=%2Ftree%3F (192.168.8.122) 1.00ms
[I 08:29:23.079 NotebookApp] Kernel started: 9c4d7d86-5dfd-4b94-8eda-1e2f9159a5c0
[I 08:29:23.644 NotebookApp] Adapting to protocol v5.1 for kernel 9c4d7d86-5dfd-4b94-8eda-1e2f9159a5c0
GLFW Error: X11: The DISPLAY environment variable is missing
Failed to initialize GLFW
[DrawGeometries] Failed creating OpenGL window.
```

======

sorry you have already ask the same question on Open3D

https://github.com/IntelVCL/Open3D/issues/376

**Comment by adioshun** [comment]
@wooheaven I also have the same problem(It looks the docker dependency problem) that is why I commemt the last line of jupyter file. Why don't you install the Open3D on you computer. it is quite easy.

```
Ubuntu 16 & 18, python3, pip
apt-get install cmake 
pip3 install numpy

cd ~
git clone https://github.com/IntelVCL/Open3D

# install the dependencies and use CMake to build the project
cd ~/Open3D
util/scripts/install-deps-ubuntu.sh
mkdir build
cd build
cmake ../src
make -j

# Install Python binding module 

cd util/scripts
./install.sh
```

I checked it and it works

**Comment by wooheaven** [comment]
Thank you !! @adioshun

**Comment by bigeyesung** [comment]
Hey I want to follow up this question.
I have the same one when I run in docker image on aws EC2 instance.
I can't run it locally due to memory is not enough.
`apt-get install libglfw3-dev`
It didn't work for me

**Comment by haoliangjiang** [comment]
> Hey I want to follow up this question.
> I have the same one when I run in docker image on aws EC2 instance.
> I can't run it locally due to memory is not enough.
> `apt-get install libglfw3-dev`
> It didn't work for me

I am having the same issue when working on a remote machine. The package is installed using "RUN pip install open3d" in my dockerfile. It is something related to Xforwarding?

**Comment by malsaidi93** [comment]
I want to visualize pointcloud from RGBD image. Followed the tutorial on open3d website and I'm getting this error
pcd.transform([[1, 0, 0, 0], [0, -1, 0, 0], [0, 0, -1, 0], [0, 0, 0, 1]])
     PointCloud with 102400 points.
o3d.visualization.draw_geometries([pcd])
[Open3D WARNING] GLFW Error: X11: The DISPLAY environment variable is missing
[Open3D WARNING] Failed to initialize GLFW
[Open3D WARNING] [DrawGeometries] Failed creating OpenGL window.


I'm running my notebook on aws EC2 isntance 
open3d was installed via pip (today)

**Comment by anyaviswa** [comment]
I am getting the same error when I try visualizing image using open3d on google colab, any help on how to resolve this is appreciated.
[Open3D WARNING] GLFW Error: X11: The DISPLAY environment variable is missing0;m
[Open3D WARNING] Failed to initialize GLFW

**Comment by AliKaramiFBK** [comment]
I have the same issue.
hope someone can help

**Comment by 1LOVESJohnny** [comment]
Same problem here.

**Comment by Daniel22L** [comment]
Same problem 
I have followed this instructions:
http://www.open3d.org/docs/release/docker.html

I would like to use the Open3D viewer application in python running in a Docker container.

**Comment by ossorio** [comment]
Recently, I have installed and configured a jupyterhub server for collaborative coding and execution of python code taking use of a GPU. I managed to find a solution for the "DISPLAY environment variable is missing" error that you are referring to.

Basically, it involves XForwarding, as @haoliangjiang suggested.

**1. First alternative:** (**Linux client machine** connecting to Linux server with Jupyter notebook/Jupyter Lab/Jupyter Hub).

Firstly, you need to install  [Xephyr](https://wiki.archlinux.org/title/Xephyr) in the server side. Xephyr provides the ability to create a X server nested under another X server. 

Afterwards, you need to open an SSH connection to the server **with X11 Forwarding enabled (use -X ssh option)** and execute Xephyr, for example: "_Xephyr -br -ac -noreset -screen 1920x1080 :99_". The **:99** is the virtual display which you are creating, so we will use this reference later.
When you execute Xephyr this way, a new black window will appear in your desktop. Don't worry, it is the virtual display.

Now we have to focus on the python code in the remote server in Jupyter Lab/notebook/Hub. Before the portions of code that make use of a virtual display (drawings, Visualizer, etc.) it is compulsory to add our display environment variable ID. For example: _os.environ['DISPLAY'] = ':99.0'_. **This way the program will draw the graphics in our local window**. 

As a detail, if you want to use the web_visualizer of Open3D, you must remove the DISPLAY environment variable from the program (otherwise, at least in my experience, the kernel may crash). For example: _del os.environ['DISPLAY']_

**2. Second alternative** (**Windows client machine** connecting to Linux server with Jupyter notebook/Jupyter Lab/Jupyter Hub).

Basically, you have to do the same as in the first alternative, but previously you must install an Xserver in your Windows machine (Vcxsrv or Xming, for example). You should also install a SSH client and configure the X11 Forwarding adding the DISPLAY generated by the Xserver (**:0.0** by default with Vcxsrv).

**3. Third alternative** (Use plotly to generate 3D visualizations)
I do not know very much about this library, but it allows to create visualizations in the web browser without having to mess with the virtual display problem.
[Plotly](https://plotly.com/python/)

**Comment by ramers87** [comment]
> @wooheaven I also have the same problem(It looks the docker dependency problem) that is why I commemt the last line of jupyter file. Why don't you install the Open3D on you computer. it is quite easy.
> 
> ```
> Ubuntu 16 & 18, python3, pip
> apt-get install cmake 
> pip3 install numpy
> 
> cd ~
> git clone https://github.com/IntelVCL/Open3D
> 
> # install the dependencies and use CMake to build the project
> cd ~/Open3D
> util/scripts/install-deps-ubuntu.sh
> mkdir build
> cd build
> cmake ../src
> make -j
> 
> # Install Python binding module 
> 
> cd util/scripts
> ./install.sh
> ```
> 
> I checked it and it works

 i faced problem in src file it's not existed how i can solve it please

**Comment by ichsan2895** [comment]
> > @wooheaven I also have the same problem(It looks the docker dependency problem) that is why I commemt the last line of jupyter file. Why don't you install the Open3D on you computer. it is quite easy.
> > ```
> > Ubuntu 16 & 18, python3, pip
> > apt-get install cmake 
> > pip3 install numpy
> > 
> > cd ~
> > git clone https://github.com/IntelVCL/Open3D
> > 
> > # install the dependencies and use CMake to build the project
> > cd ~/Open3D
> > util/scripts/install-deps-ubuntu.sh
> > mkdir build
> > cd build
> > cmake ../src
> > make -j
> > 
> > # Install Python binding module 
> > 
> > cd util/scripts
> > ./install.sh
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I checked it and it works
> 
> i faced problem in src file it's not existed how i can solve it please

I also faced this problem and solved by using `cmake ..` instead of `cmake ../src` in `build` directory,

**Comment by ramers87** [comment]
HiTry this command to solve the variable problemEnvoyé depuis mon mobile Huawei-------- Message original --------Objet : Re: [adioshun/gitBook_PCL] I got a Error = [ GLFW Error: X11: The DISPLAY environment variable is missing ] (#1)De : Muhammad Ichsan À : adioshun/gitBook_PCL Cc : Rami Merzoug ,Comment 


@wooheaven I also have the same problem(It looks the docker dependency problem) that is why I commemt the last line of jupyter file. Why don't you install the Open3D on you computer. it is quite easy.
Ubuntu 16 & 18, python3, pip
apt-get install cmake 
pip3 install numpy

cd ~
git clone https://github.com/IntelVCL/Open3D

# install the dependencies and use CMake to build the project
cd ~/Open3D
util/scripts/install-deps-ubuntu.sh
mkdir build
cd build
cmake ../src
make -j

# Install Python binding module 

cd util/scripts
./install.sh

I checked it and it works

i faced problem in src file it's not existed how i can solve it please

I also faced this problem and solved by using cmake .. instead of cmake ../src in build directory,

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/adioshun/gitBook_PCL/issues/1#issuecomment-1365512028",
"url": "https://github.com/adioshun/gitBook_PCL/issues/1#issuecomment-1365512028",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by SandeepaDevin** [comment]
Running this same issue in kaggle notebooks

[Open3D WARNING] GLFW Error: X11: The DISPLAY environment variable is missing
[Open3D WARNING] Failed to initialize GLFW
[Open3D WARNING] [DrawGeometries] Failed creating OpenGL window.

**Comment by Ajaypawar02** [comment]
i am also facing the same issue while i am running this on ec2 instance . Any suggestion how i can debug this error

**Comment by mariahlusneac** [comment]
https://stackoverflow.com/questions/74774761/problem-with-x-display-when-open3d-draw-geometry-used
The solution provided here solved the issue for me - using `draw_plotly` instead of `draw_geometries`.

**Comment by dakies-sony** [comment]
`draw_plotly `crashes the kernel for me

**Comment by justinclimendraj** [comment]
Timestamp,Level,Message
"Sep 6, 2023, 12:01:29 PM",WARNING,WARNING:root:kernel cd822a92-d3bc-4985-bac1-2f2345cb4c97 restarted
"Sep 6, 2023, 12:01:29 PM",INFO,"KernelRestarter: restarting kernel (1/5), keep random ports"
"Sep 6, 2023, 11:49:38 AM",WARNING,WARNING:root:kernel cd822a92-d3bc-4985-bac1-2f2345cb4c97 restarted
"Sep 6, 2023, 11:49:38 AM",INFO,"KernelRestarter: restarting kernel (1/5), keep random ports"
"Sep 6, 2023, 11:48:14 AM",WARNING,WARNING:root:kernel cd822a92-d3bc-4985-bac1-2f2345cb4c97 restarted
"Sep 6, 2023, 11:48:14 AM",INFO,"KernelRestarter: restarting kernel (1/5), keep random ports"
"Sep 6, 2023, 11:44:17 AM",WARNING,WARNING:root:kernel cd822a92-d3bc-4985-bac1-2f2345cb4c97 restarted
"Sep 6, 2023, 11:44:17 AM",INFO,"KernelRestarter: restarting kernel (1/5), keep random ports"
"Sep 6, 2023, 11:42:14 AM",WARNING,WARNING:root:kernel cd822a92-d3bc-4985-bac1-2f2345cb4c97 restarted
"Sep 6, 2023, 11:42:14 AM",INFO,"KernelRestarter: restarting kernel (1/5), keep random ports"
"Sep 6, 2023, 11:38:59 AM",WARNING,WARNING:root:kernel cd822a92-d3bc-4985-bac1-2f2345cb4c97 restarted
"Sep 6, 2023, 11:38:59 AM",INFO,"KernelRestarter: restarting kernel (1/5), keep random ports"
"Sep 6, 2023, 11:38:11 AM",WARNING,WARNING:root:kernel cd822a92-d3bc-4985-bac1-2f2345cb4c97 restarted
"Sep 6, 2023, 11:38:11 AM",INFO,"KernelRestarter: restarting kernel (1/5), keep random ports"
"Sep 6, 2023, 11:37:38 AM",WARNING,WARNING:root:kernel cd822a92-d3bc-4985-bac1-2f2345cb4c97 restarted
"Sep 6, 2023, 11:37:38 AM",INFO,"KernelRestarter: restarting kernel (1/5), keep random ports"
"Sep 6, 2023, 11:36:38 AM",WARNING,WARNING:root:kernel cd822a92-d3bc-4985-bac1-2f2345cb4c97 restarted
"Sep 6, 2023, 11:36:38 AM",INFO,"KernelRestarter: restarting kernel (1/5), keep random ports"
"Sep 6, 2023, 11:35:53 AM",WARNING,WARNING:root:kernel cd822a92-d3bc-4985-bac1-2f2345cb4c97 restarted
"Sep 6, 2023, 11:35:53 AM",INFO,"KernelRestarter: restarting kernel (1/5), keep random ports"
"Sep 6, 2023, 11:27:02 AM",INFO,"Kernel started: cd822a92-d3bc-4985-bac1-2f2345cb4c97, name: python3"
"Sep 6, 2023, 11:20:30 AM",INFO,Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
"Sep 6, 2023, 11:20:30 AM",INFO,http://172.28.0.12:9000/
"Sep 6, 2023, 11:20:30 AM",INFO,Jupyter Notebook 6.5.5 is running at:
"Sep 6, 2023, 11:20:30 AM",INFO,Serving notebooks from local directory: /
"Sep 6, 2023, 11:20:30 AM",INFO,Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
"Sep 6, 2023, 11:20:30 AM",INFO,http://172.28.0.2:9000/
"Sep 6, 2023, 11:20:30 AM",INFO,Jupyter Notebook 6.5.5 is running at:
"Sep 6, 2023, 11:20:30 AM",INFO,Serving notebooks from local directory: /
"Sep 6, 2023, 11:20:25 AM",INFO,google.colab serverextension initialized.
"Sep 6, 2023, 11:20:25 AM",INFO,google.colab serverextension initialized.
"Sep 6, 2023, 11:20:25 AM",INFO,"Authentication of /metrics is OFF, since other authentication is disabled."
"Sep 6, 2023, 11:20:25 AM",INFO,Writing notebook server cookie secret to /root/.local/share/jupyter/runtime/notebook_cookie_secret
"Sep 6, 2023, 11:20:25 AM",WARNING,    	/root/.jupyter/jupyter_notebook_config.json
"Sep 6, 2023, 11:20:25 AM",WARNING,    	/root/.local/etc/jupyter/jupyter_notebook_config.json
"Sep 6, 2023, 11:20:25 AM",WARNING,    	/usr/etc/jupyter/jupyter_notebook_config.json
"Sep 6, 2023, 11:20:25 AM",WARNING,    	/usr/local/etc/jupyter/jupyter_notebook_config.json
"Sep 6, 2023, 11:20:25 AM",WARNING,    	/usr/local/etc/jupyter/jupyter_notebook_config.d/panel-client-jupyter.json
"Sep 6, 2023, 11:20:25 AM",WARNING,    	/etc/jupyter/jupyter_notebook_config.json
"Sep 6, 2023, 11:20:25 AM",INFO,"Authentication of /metrics is OFF, since other authentication is disabled."
"Sep 6, 2023, 11:20:25 AM",INFO,Writing notebook server cookie secret to /root/.local/share/jupyter/runtime/notebook_cookie_secret
"Sep 6, 2023, 11:20:25 AM",WARNING,    	/root/.jupyter/jupyter_notebook_config.json
"Sep 6, 2023, 11:20:25 AM",WARNING,    	/root/.local/etc/jupyter/jupyter_notebook_config.json
"Sep 6, 2023, 11:20:25 AM",WARNING,    	/usr/etc/jupyter/jupyter_notebook_config.json
"Sep 6, 2023, 11:20:25 AM",WARNING,    	/usr/local/etc/jupyter/jupyter_notebook_config.json
"Sep 6, 2023, 11:20:25 AM",WARNING,    	/usr/local/etc/jupyter/jupyter_notebook_config.d/panel-client-jupyter.json
"Sep 6, 2023, 11:20:25 AM",WARNING,    	/etc/jupyter/jupyter_notebook_config.json
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.368 NotebookApp] Loaded config file: /root/.jupyter/jupyter_notebook_config.py
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.368 NotebookApp] Loaded config file: /root/.jupyter/jupyter_notebook_config.py
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.364 NotebookApp] Looking for jupyter_notebook_config in /root/.jupyter
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.364 NotebookApp] Looking for jupyter_notebook_config in /root/.local/etc/jupyter
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.364 NotebookApp] Looking for jupyter_notebook_config in /usr/etc/jupyter
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.364 NotebookApp] Loaded config file: /usr/local/etc/jupyter/jupyter_notebook_config.json
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.361 NotebookApp] Looking for jupyter_notebook_config in /usr/local/etc/jupyter
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.361 NotebookApp] Loaded config file: /etc/jupyter/jupyter_notebook_config.py
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.359 NotebookApp] Looking for jupyter_notebook_config in /etc/jupyter
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.364 NotebookApp] Looking for jupyter_notebook_config in /root/.jupyter
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.364 NotebookApp] Looking for jupyter_notebook_config in /root/.local/etc/jupyter
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.364 NotebookApp] Looking for jupyter_notebook_config in /usr/etc/jupyter
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.364 NotebookApp] Loaded config file: /usr/local/etc/jupyter/jupyter_notebook_config.json
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.362 NotebookApp] Looking for jupyter_notebook_config in /usr/local/etc/jupyter
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.362 NotebookApp] Loaded config file: /etc/jupyter/jupyter_notebook_config.py
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.361 NotebookApp] Looking for jupyter_notebook_config in /etc/jupyter
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.359 NotebookApp] Looking for jupyter_config in /root/.jupyter
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.359 NotebookApp] Looking for jupyter_config in /root/.local/etc/jupyter
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.358 NotebookApp] Looking for jupyter_config in /usr/etc/jupyter
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.356 NotebookApp] Looking for jupyter_config in /root/.jupyter
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.356 NotebookApp] Looking for jupyter_config in /root/.local/etc/jupyter
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.355 NotebookApp] Looking for jupyter_config in /usr/etc/jupyter
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.342 NotebookApp] Looking for jupyter_config in /usr/local/etc/jupyter
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.340 NotebookApp] Looking for jupyter_config in /etc/jupyter
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.343 NotebookApp] Looking for jupyter_config in /usr/local/etc/jupyter
"Sep 6, 2023, 11:20:24 AM",WARNING,[D 05:50:24.343 NotebookApp] Looking for jupyter_config in /etc/jupyter
"Sep 6, 2023, 11:20:24 AM",WARNING,"[D 05:50:24.340 NotebookApp] Searching ['/root/.jupyter', '/root/.local/etc/jupyter', '/usr/etc/jupyter', '/usr/local/etc/jupyter', '/etc/jupyter'] for config files"
"Sep 6, 2023, 11:20:24 AM",WARNING,"[D 05:50:24.340 NotebookApp] Searching ['/root/.jupyter', '/root/.local/etc/jupyter', '/usr/etc/jupyter', '/usr/local/etc/jupyter', '/etc/jupyter'] for config files"






i got this error some give  idea to solve this

**Comment by liuxinhai** [comment]
> I am getting the same error when I try visualizing image using open3d on google colab, any help on how to resolve this is appreciated. [Open3D WARNING] GLFW Error: X11: The DISPLAY environment variable is missing0;m [Open3D WARNING] Failed to initialize GLFW

You may need to ser the DISPLAY value, such as `export DISPLAY=:0.0` or `export DISPLAY=:99.0`.

**Comment by ramers87** [comment]
Hi, The solution is written down in comments, you didn't see it? Envoyé depuis mon mobile Huawei-------- Message original --------Objet : Re: [adioshun/gitBook_PCL] I got a Error = [ GLFW Error: X11: The DISPLAY environment variable is missing ] (#1)De : Xinhai Liu À : adioshun/gitBook_PCL Cc : Rami Merzoug ,Comment 

I am getting the same error when I try visualizing image using open3d on google colab, any help on how to resolve this is appreciated. [Open3D WARNING] GLFW Error: X11: The DISPLAY environment variable is missing0;m [Open3D WARNING] Failed to initialize GLFW

You may need to ser the DISPLAY value, such as ``export DISPLAY=:0.0or `export DISPLAY=:99.0`.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/adioshun/gitBook_PCL/issues/1#issuecomment-1718723407",
"url": "https://github.com/adioshun/gitBook_PCL/issues/1#issuecomment-1718723407",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by benlam62** [comment]
> Recently, I have installed and configured a jupyterhub server for collaborative coding and execution of python code taking use of a GPU. I managed to find a solution for the "DISPLAY environment variable is missing" error that you are referring to.
> 
> Basically, it involves XForwarding, as @haoliangjiang suggested.
> 
> **1. First alternative:** (**Linux client machine** connecting to Linux server with Jupyter notebook/Jupyter Lab/Jupyter Hub).
> 
> Firstly, you need to install [Xephyr](https://wiki.archlinux.org/title/Xephyr) in the server side. Xephyr provides the ability to create a X server nested under another X server.
> 
> Afterwards, you need to open an SSH connection to the server **with X11 Forwarding enabled (use -X ssh option)** and execute Xephyr, for example: "_Xephyr -br -ac -noreset -screen 1920x1080 :99_". The **:99** is the virtual display which you are creating, so we will use this reference later. When you execute Xephyr this way, a new black window will appear in your desktop. Don't worry, it is the virtual display.
> 
> Now we have to focus on the python code in the remote server in Jupyter Lab/notebook/Hub. Before the portions of code that make use of a virtual display (drawings, Visualizer, etc.) it is compulsory to add our display environment variable ID. For example: _os.environ['DISPLAY'] = ':99.0'_. **This way the program will draw the graphics in our local window**.
> 
> As a detail, if you want to use the web_visualizer of Open3D, you must remove the DISPLAY environment variable from the program (otherwise, at least in my experience, the kernel may crash). For example: _del os.environ['DISPLAY']_
> 
> **2. Second alternative** (**Windows client machine** connecting to Linux server with Jupyter notebook/Jupyter Lab/Jupyter Hub).
> 
> Basically, you have to do the same as in the first alternative, but previously you must install an Xserver in your Windows machine (Vcxsrv or Xming, for example). You should also install a SSH client and configure the X11 Forwarding adding the DISPLAY generated by the Xserver (**:0.0** by default with Vcxsrv).
> 
> **3. Third alternative** (Use plotly to generate 3D visualizations) I do not know very much about this library, but it allows to create visualizations in the web browser without having to mess with the virtual display problem. [Plotly](https://plotly.com/python/)

Thanks but there is no new black window appear when execute Xephyr.  Do you know why?

**Comment by Gaozihui** [comment]
> https://stackoverflow.com/questions/74774761/problem-with-x-display-when-open3d-draw-geometry-used The solution provided here solved the issue for me - using `draw_plotly` instead of `draw_geometries`.

work for me！
